presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -305,6 +305,20 @@ private void processNewSources()
         SettableFuture<?> newDriverBlockedFuture = SettableFuture.create();
         driverBlockedFuture.set(newDriverBlockedFuture);
         sourceBlockedFuture.addListener(() -> newDriverBlockedFuture.set(null), newDirectExecutorService());
+
+        // it's possible that memory revoking is requested for some operator
+        // before we update driverBlockedFuture above and we don't want to miss that
+        // notification, so we check to see whether that's the case before returning.
+        boolean memoryRevokingRequested = operators.stream()
+                .map(Operator::getOperatorContext)
+                .filter(OperatorContext::isMemoryRevokingRequested)
+                .findAny()
+                .isPresent();
+
+        if (memoryRevokingRequested) {
+            newDriverBlockedFuture.set(null);
+        }
```

Comment:
Would be nice to have a test for that.

@findepi came with idea to test it with hand crafted `Driver` instance, with an operator which always returns blocked future from `getOutput` and within `getOutput` it requests  memory revoking on itself.

Reply:
some questions/issues:
- if an operator is always blocked its getOutput will never be called. (I guess we can request revocation in isBlocked())
- if an operator is blocked its finish() will never be called, so we cannot write a test that loops until the driver is finished as it will never happen. Should we iterate N times, will that be useful?

Here is a test that never completes as finish is never called: https://gist.github.com/nezihyigitbasi/0ff5e28f8dab70a460cee3b5316607c3

@findepi can you give more details about what you have in mind?